group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -620,9 +637,9 @@ public Map<String, TopicMetadata> computeSubscriptionMetadata(
         TopicsImage topicsImage,
         ClusterImage clusterImage
     ) {
-        // Copy and update the current subscriptions.
+        // Copy and update the current subscription information.
         Map<String, Integer> subscribedTopicNames = new HashMap<>(this.subscribedTopicNames);
-        maybeUpdateSubscribedTopicNames(subscribedTopicNames, oldMember, newMember);
+        maybeUpdateSubscribedTopicNamesAndGroupSubscriptionModel(subscribedTopicNames, oldMember, newMember);
```

Comment:
Hum... We need to be careful here because  we are not suppose to update the internal state of the group on this code path. It may be better to keep it as it was before and to have another method to update the subscription type.